### ISSUE TITLE
BUG: `_var` in distributions discrete looks wrong

### ISSUE THREAD

**Issue by josef-pkt** [statsmodels/statsmodels#7161]
**update, status** vectorization fixed, but needs more work to be consistent with scipy distributions, see last comment

see #3887 for SUMM issue

I think there should not be a `.mean()` in the var computation
e.g. for ZIP, but similar for other `distributions.discrete`

```
    def _var(self, mu, w):
        dispersion_factor = 1 + w * mu
        var = (dispersion_factor * self._mean(mu, w)).mean()
        return var
```

I guess we want to compute vectorized `var` for all parameter combinations

unit test in distributions only uses scalar params

**Comment by josef-pkt** [comment]
missing unit test coverage for mean, var and moments
https://coveralls.io/builds/35742790/source?filename=statsmodels/distributions/discrete.py

**Comment by josef-pkt** [comment]
formula for var of zip might also be wrong
https://stats.stackexchange.com/questions/18661/mean-and-variance-of-a-zero-inflated-poisson-distribution
(I'm too impatient right now to check the math)

**Comment by josef-pkt** [comment]
vectorized version is fixed by PR #7281 

but `_mean`, `_var` doesn't really fit in scipy distribution framework, leaving this issue open

copying my comments in PR:

However,
The scipy distribution framework does **not** use `_mean` and `_var`, so calling `mean` and `var` still goes through the generic path.

In the scipy version that I use, moments, stats, mean, var are not vectorized.
AFAICS, in scipy master they should be vectorized, at least mean, var and stats.
This will need a proper solution, either we define `mean` and `var` without going through scipy, or we define  mean and var in `_stats` which might be vectorized for some recent scipy.
Changing this will need checking with different scipy versions.

**Comment by josef-pkt** [comment]
This is now finished. (except I didn't check what generic distribution `stats` method uses)
#7929 (for 0.14) renamed to drop underline in mean and var. unit tested with new `get_distribution` model methods

BUG for underline function was already fixed in 0.13,
